Appsilon/rhino
# Configure pre-commit

I was working on another thing and didn't want to figure out what checks to run, so... I configured pre-commit 😁

## Description
This PR adds [pre-commit](https://pre-commit.com) config. pre-commit introduces a series of checks (and fixes) that can be run as git hooks. It is loosely based on [a config that I used in shiny.telemetry](https://github.com/Appsilon/shiny.telemetry/pull/140). I have improved the config to not require globally installed packages.

Note: I have specifically omitted running tests as they can get slow with time and running pre-commit on a pre-commit hook would get annoying.

### How to Test
1. Follow setup instructions in CONTRIBUTING.md
2. Run `pre-commit run --all-files` to see if the checks pass

### Proposed Next Steps
1. Configure [pre-commit.ci](https://pre-commit.ci)
1. Remove duplication of checks between pre-commit.ci and GitHub Actions

## Definition of Done
- [x] The change is thoroughly documented.
- [x] The CI passes (`R CMD check`, linter, unit tests, spelling).
- [x] Any generated files have been updated (e.g. `.Rd` files with `roxygen2::roxygenise()`)
